### PR TITLE
feat(apigateway): configure integration options on LambdaRestApi

### LINKED ISSUE #3269: ApiGateway LambdaRestApi doesn't allow overriding default integration properties
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* **I'm submitting a ...**
  - [ ] :beetle: bug report
  - [X] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/awslabs/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

When defining LambdaRestApi I am unable to assign credentials role to the lambda. Only handler can be passed: https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts#L51

* **What is the expected behavior (or behavior of feature suggested)?**

I should be able to assign credentialsRole or other integration properties.

* **What is the motivation / use case for changing the behavior or adding this feature?**

When users are introduced to apigateway constructs are going to choose lambda rest api naturally when having a lambda handler.  However they will soon find they can't override any props on the integration (https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts#L46) and will need to switch to a regular RestApi.

* **Please tell us about your environment:**
  
  - CDK CLI Version: 0.38.0
  - Module Version: 0.38.0
  - OS: All
  - Language: All

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

### PR DESCRIPTION
**Pull request by yamatatsu**
close #3269

But I wonder that why #3269 was removed label "good first issue".

I guess there are some plans to resolve this issue without fix as this PR?
Assuming that, I'm sorry that I try to fix this issue without your consultation.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts [hunk 1]
```diff
@@ -14,6 +14,13 @@ export interface LambdaRestApiProps extends RestApiProps {
    */
   readonly handler: lambda.IFunction;
 
+  /**
+   * Integration options, such as request/resopnse mapping, content handling
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts:18]:
```suggestion
   * Integration options, such as request/response mapping, content handling
```

File: packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts [hunk 2]
```diff
@@ -14,6 +14,13 @@ export interface LambdaRestApiProps extends RestApiProps {
...
+   *
+   * @default - no options.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts:20]:
```suggestion
   * @default - see defaults defined in `LambdaIntegrationOptions`.
```

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. Please see some of my comments below.

File: packages/@aws-cdk/aws-apigateway/README.md [hunk 3]
```diff
@@ -20,31 +20,33 @@ running on AWS Lambda, or any web application.
 
 ## Table of Contents
 
-- [Defining APIs](#defining-apis)
```
[the remark this answers is not in the document]
Comment (yamatatsu) [reply packages/@aws-cdk/aws-apigateway/README.md]: Oh. OK. I did not intend.

File: packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts (hunk 1 above)
Comment (yamatatsu) [reply packages/@aws-cdk/aws-apigateway/lib/lambda-api.ts]: lol🤣

**Comment by yamatatsu** [comment]
Thanks for your comments, nija-at.
I believe I've addressed all comments.

**Review by nija-at — APPROVED** [review]

**Review by nija-at — CHANGES_REQUESTED** [review]
Looks like there's a failing test. Can you look at the build output. A link to the build logs are in the automated comment above.

**Review by nija-at — CHANGES_REQUESTED** [review]
Could you address the failing build?

**Comment by nija-at** [comment]
Closing this old pull request. Please open a new pull request when you have updates.

**Pull request closed without merging**